docs/tests-ui.rst

```diff
@@ -1,177 +1,148 @@
-Client-side Testing with Intern
-===============================
-From 2014 to 2016, we used Intern_ for client-side testing. It uses Selenium
-WebDriver API which lets us write automated testing via JavaScript. Intern is
-an open source project created and maintained by SitePen_.
+===================
+Client-side Testing
+===================
 
-In 2016, we started to convert our client-side tests to Python, to standardize
-on common Mozilla QA processes, and as part of an effort to add client-side
-tests to the development pipline.  The Intern tests described here are no
-longer run for development and deployment, and have not been updated for the
-Docker development environment.
+Kuma has a suite of functional tests using `Selenium`_ and `pytest`_. This allows us
+to emulate users interacting with a real browser. All these test suites live in
+the /tests/ directory.
 
-.. _SitePen: http://sitepen.com
-.. _Intern: https://theintern.github.io/
+The tests directory comprises of:
 
-Installing Dependencies
------------------------
+   * ``/functional`` contains pytest tests.
+   * ``/pages`` contains Python pypom page objects.
+   * ``/utils`` contains helper functions.
 
-1. Go to the ``tests/ui/`` directory::
+.. _`Selenium`: http://docs.seleniumhq.org/
+.. _`pytest`: http://pytest.org/latest/
 
-    cd tests/ui
+Setting up test virtual environment
+===================================
 
-2. Use ``npm`` to install Intern::
+#. In the terminal go to the directory where you have downloaded kuma
 
-    npm install intern@^3.0
+#. Create a virtual environment for the tests to run in (the example uses the
+   name mdntests)::
 
-.. warning:: Do *not* install Intern globally -- path issues may occur.
+   $ virtualenv mdntests
 
-Running Tests
--------------
+#. Activate the virtual environment::
 
-On your machine
-~~~~~~~~~~~~~~~
+   $ source mdntests/bin/activate
 
-1. Install `JDK <http://www.oracle.com/technetwork/java/javase/downloads/index.html>`_
+#. Install the requirements the tests need to run::
 
-2. Download the most current `release of Selenium <http://selenium-release.storage.googleapis.com/index.html>`_ standalone server. (It's the ``.jar`` file.)
+   $ pip install -r requirements/test.txt
```

Comment:
I had to upgrade ``pip`` as well, since my default is 7.1.2, which is before hashes.  This may just be me and my old laptop.  To enforce it, add:

    $ pip install "pip>=8"